### DISCUSSION TITLE
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

### DISCUSSION THREAD

**Discussion by ghost** [webpack/webpack#17819] [Q&A] [answered]
A number of developers have recently been getting the above warning from `webpack` due to a `supabase` -> `websockets` -> `node-gyp-build` dependency

The code in `node-pre-gyp` that's raising the warning is a conditional require statement:
```
if (typeof require.addon === 'function') { // if the platform supports native resolving prefer that
  module.exports = require.addon.bind(require)
} else { // else use the runtime version here
  module.exports = require('./node-gyp-build.js')
}
```

The issue has been discussed here: https://github.com/prebuild/node-gyp-build/issues/62
And here: https://github.com/supabase/realtime-js/issues/265#issuecomment-1820546931

The `node-gyp-build` team have an internal javascript runtime that requires the conditional.

The question I have is: is it a requirement of webpack to not use conditionals around require statements?

Here's a link to the warning: in `webpack`:
https://github.com/webpack/webpack/blob/87660921808566ef3b8796f8df61bd79fc026108/lib/dependencies/CommonJsImportsParserPlugin.js#L216

I looked for the code where that configuration flag is set, but was not able to locate, so it wasn't clear based on the code. Please let us know. Thanks!

@kasperisager @mafintosh @filipecabaco

**Comment by ghost** [comment]
@vankop for visibility

**Comment by tim-vyg** [comment]
Currently experiencing this as well!

**Comment by AndyGaSa** [comment]
I'm having the same issue sadly

**Comment by Ali-Onar** [comment, marked as answer]
I updated the supabase package yesterday. I've been getting this error several times today. 

Old version: `"@supabase/supabase-js": "^2.36.0"`
New version: `"@supabase/supabase-js": "^2.38.5"`

When I installed the old package again, this error persisted. 
So I also restored the package-lock.json file. 
After running the `npm install` command, the problem was solved.
